packages/file-storage/src/file-storage-s3.class.ts

```diff
@@ -19,29 +20,41 @@ import {
   FileStorageDirBaseArgs,
 } from './file-storage.class';
 
+/**
+ * Either region or endpoint must be provided
+ */
 export type FileStorageS3Setup = {
-  accessKeyId: string;
   bucket: string;
-  endpoint: string;
-  secretAccessKey?: string;
-  s3BucketEndpoint?: boolean;
   maxPayloadSize: number;
+  credentials: {
+    accessKeyId: string;
+    secretAccessKey: string;
+  };
   [key: string]: unknown;
-};
+} & ({ region: string } | { endpoint: string });
 
 export interface FileStorageS3Config {
   s3: S3;
   bucket: string;
   [key: string]: any;
 }
 
+export function extractRegionFromEndpoint(endpoint: string): string {
+  const error = new Error('Please add a valid region or endpoint');
+  if (!endpoint) throw error;
+
+  const match = endpoint.match(/(?<=\.)[^.]+(?=\.amazonaws\.com)/);
+  if (match) {
+    return match[0];
+  }
+  throw error;
```

Comment:
are you sure this error is needed ?
What happens if you pass an invalid region to S3 class constructor ?

Reply:
If you pass an invalid region, it throws an AWS error like this one:
```
AWS SDK error wrapper for Error: getaddrinfo ENOTFOUND s1seven-schema-service-test-local.s3.eu-central-10.amazonaws.com
```
Should I get a list of valid regions and validate it in the function? I didn't do that initially as I thought it would add more long term maintenance as the list would have to be kept up to date.

Reply:
good then let's make your code simpler and `return match?.length ? match[0] : null;`